identity/src/test/java/com/android/identity/SessionEncryptionInteractionTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.identity;
+
+import androidx.core.util.Pair;
+import androidx.test.filters.SmallTest;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.OptionalLong;
+import java.util.Random;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.model.Array;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.DataItem;
+
+public class SessionEncryptionInteractionTest {
```

Comment:
I was wondering why this is a separate file and then I realized it's because this uses both the reader and device classes, which makes sense.

I think we should actually combine `SessionEncryptionDevice` and `SessionEncryptionReader` into a single `SessionEncryption` since there's a bunch of duplication between the two. Its constructor would take take a `role` parameter (which can be `ROLE_MDOC` and `ROLE_MDOC_READER`) much the same way our `DataTransport` abstraction works (at some point we can combine the constants/IntDef, suggest to keep separate for now).

This would we would just have a single `SessionEncryptionTest` file instead of the three we ends up with after this PR (`SessionEncryptionDeviceTest`, `SessionEncryptionReaderTest`, `SessionEncryptionInteractionTest`).

I realize this is a bit of extra work but I think it goes a long way to make the code easier to read!